src/main/fc/rc.c

```diff
@@ -489,141 +492,177 @@ static FAST_CODE void processRcSmoothingFilter(void)
 }
 #endif // USE_RC_SMOOTHING_FILTER
 
-NOINLINE void initAveraging(uint16_t feedforwardAveraging)
-{
-    for (int i = 0; i < XYZ_AXIS_COUNT; i++) {
-        laggedMovingAverageInit(&feedforwardDeltaAvg[i].filter, feedforwardAveraging + 1, (float *)&feedforwardDeltaAvg[i].buf[0]);
-    }
-}
-
-FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, int axis)
+#ifdef USE_FEEDFORWARD
+FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, flight_dynamics_index_t axis)
 {
     const float rxInterval = currentRxIntervalUs * 1e-6f; // seconds
-    float rxRate = currentRxRateHz;
-    static float prevRxInterval;
-
-    static float prevRcCommand[3];
+    float rxRate = currentRxRateHz;                 // 1e6f / currentRxIntervalUs;
+    static float prevRcCommand[3];                  // for rcCommandDelta test
     static float prevRcCommandDeltaAbs[3];          // for duplicate interpolation
-    static float prevSetpoint[3];                   // equals raw unless interpolated 
-    static float prevSetpointSpeed[3];
-    static float prevAcceleration[3];               // for duplicate interpolation
-    static bool prevDuplicatePacket[3];             // to identify multiple identical packets
-    static uint16_t feedforwardAveraging = 0;
-
-    if (feedforwardAveraging != pid->feedforwardAveraging) {
-        feedforwardAveraging = pid->feedforwardAveraging;
-        initAveraging(feedforwardAveraging);
-    }
+    static float prevSetpoint[3];                   // equals raw unless extrapolated forward 
+    static float prevSetpointSpeed[3];              // for setpointDelta calculation
+    static float prevSetpointSpeedDelta[3];         // for duplicate extrapolation
+    static bool prevPacketWasADuplicate[3];             // to identify multiple identical packets
```

Comment:
I'm still confused by the name of this variable. The prefix "prev" already implies something from the past, so "was" becomes redundant. I believe "isPrevPackedDuplicate" is the most appropriate name. However, if everyone else is satisfied with the current name, I'm good with it too - This isn’t as big of an issue as the discussion might suggest.

Reply:
OK alright I give in ;-)